.devcontainers/devcontainer.json

```diff
@@ -0,0 +1,41 @@
+// For format details, see https:https://containers.dev/implementors/spec/. For config options, see the README at:
+// https://github.com/devcontainers/images/tree/main/src/base-debian
+{
+    "name": "Kubernetes with e2e testing image build and push",
+    "image": "mcr.microsoft.com/devcontainers/base:bullseye",
+    // Setup the go environment and mount into the dev container at the expected location
+    "hostRequirements": {
+        "cpus": 4
+    },
+    // Copy over welcome message and install pyyaml
+    "onCreateCommand": "bash .devcontainer/setup.sh",
+    // for Kubernetes testing, suppress extraneous forwarding messages
+    "otherPortsAttributes": {
+        "onAutoForward": "silent"
+    },
+    "remoteUser": "root",
+    // Install utils, go, docker-in-docker, and other tools
+    // For more info about dev container Features see https://containers.dev/features
+    "features": {
+        "ghcr.io/devcontainers/features/go:1": {
+            "version": "1.21.1",
+            "username": "root"
+        },
+        // Install kubectl, but skip minikube and helm
```

Comment:
If we are developing against the code base, should we be building kubectl from source?

Reply:
I think that would slow down the build a lot? We want to try and get that balance right.

Happy to change that if it is a priority